backends/dnf/pk-backend-dnf.c

```diff
@@ -150,12 +151,34 @@ pk_backend_setup_dnf_context (DnfContext *context, GKeyFile *conf, const gchar *
 	dnf_context_set_cache_dir (context, cache_dir);
 	solv_dir = g_build_filename (destdir, "/var/cache/PackageKit", release_ver, "hawkey", NULL);
 	dnf_context_set_solv_dir (context, solv_dir);
-	repo_dir = g_build_filename (destdir, "/etc/yum.repos.d", NULL);
-	dnf_context_set_repo_dir (context, repo_dir);
 	lock_dir = g_build_filename (destdir, "/var/run", NULL);
 	dnf_context_set_lock_dir (context, lock_dir);
 	dnf_context_set_rpm_verbosity (context, "info");
 
+	/* Set repos and vars dirs */
+	repo_dirs = dnf_context_get_repos_dir (context);
+	if (repo_dirs[0] != NULL) {
```

Comment:
Can repo_dirs be NULL?

Reply:
In theory, it's possible. I don't know if I'd see it in practice, but I didn't think it'd hurt to have that conditional.

Reply:
No, I mean, you are checking that the first element is non-NULL, but can the whole repo_dirs be NULL? You are not checking for that.

Reply:
Hmm, that's a good question. I think it _might_ be possible, though rare. I should fix that for both `repo_dirs` and `var_dirs` checks...

Reply:
This should be fixed now.